eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/EnactSpec.hs

```diff
@@ -79,145 +100,343 @@ spec =
-            Map.fromList
-              [ (KeyHashObj khCC, EpochNo 50)
-              ]
-      prevGaidCommittee@(GovPurposeId gaidCommittee) <-
-        electCommittee
-          SNothing
-          drep
-          mempty
-          committeeMap
-      (khSPO, _, _) <- setupPoolWithStake $ Coin 42_000_000
-      logStakeDistr
-      submitYesVote_ (StakePoolVoter khSPO) gaidCommittee
-      replicateM_ 4 passEpoch
-      impAnn "Committee should be elected" $ do
-        committee <- getCommittee
-        committee `shouldBe` SJust (Committee committeeMap $ 1 %! 2)
-      pp <- getsNES $ nesEsL . curPParamsEpochStateL
-      returnAddr <- registerRewardAccount
-      gaidNoConf <-
-        submitProposal $
-          ProposalProcedure
-            { pProcReturnAddr = returnAddr
-            , pProcGovAction = NoConfidence (SJust prevGaidCommittee)
-            , pProcDeposit = pp ^. ppGovActionDepositL
-            , pProcAnchor = def
-            }
-      submitYesVote_ (StakePoolVoter khSPO) gaidNoConf
-      submitYesVote_ (DRepVoter $ KeyHashObj drep) gaidNoConf
-      replicateM_ 4 passEpoch
-      assertNoCommittee
-    it "Constitution" $ do
-      (dRep, committeeMember, _) <- electBasicCommittee
-      (govActionId, constitution) <- submitConstitution SNothing
-
-      ConwayGovState proposalsBeforeVotes enactStateBeforeVotes pulserBeforeVotes _ _ _ <-
-        getsNES $ newEpochStateGovStateL @era
-      submitYesVote_ (DRepVoter dRep) govActionId
-      submitYesVote_ (CommitteeVoter committeeMember) govActionId
-      ConwayGovState proposalsAfterVotes enactStateAfterVotes pulserAfterVotes _ _ _ <-
-        getsNES newEpochStateGovStateL
-
-      impAnn "Votes are recorded in the proposals" $ do
-        let proposalsWithVotes =
-              proposalsAddVote
-                (CommitteeVoter committeeMember)
-                VoteYes
-                govActionId
-                ( proposalsAddVote
-                    (DRepVoter dRep)
-                    VoteYes
-                    govActionId
-                    proposalsBeforeVotes
+      getTreasury `shouldReturn` initialTreasury
+      sumRewardAccounts withdrawals `shouldReturn` zero
+
+    it "Wthdrawals exceeding treasury submitted in several proposals within the same epoch" $ do
+      (drepC, committeeC, _) <- electBasicCommittee
+      modifyPParams $ ppGovActionLifetimeL .~ EpochInterval 5
+      initialTreasury <- getTreasury
+      -- generate withdrawals which individually do not exceed the treasury, but their sum does
+      numWithdrawals <- choose (1, 10)
+      withdrawals <- genWithdrawalsExceeding initialTreasury numWithdrawals 50
+
+      impAnn "submit in the same proposal, with no effect on the treasury" $ do
+        gaId <- submitTreasuryWithdrawals withdrawals
+        submitYesVote_ (DRepVoter drepC) gaId
+        submitYesVote_ (CommitteeVoter committeeC) gaId
+        passNEpochs 2
+        getTreasury `shouldReturn` initialTreasury
```

Comment:
Oh yeah. Good point. I knew I was missing something. 
Should this check be extracted into a let binding, so that both test cases can use it?
Feel free to leave it as it is.